packages/design-picker/src/hooks/use-theme-designs-query.ts

```diff
@@ -0,0 +1,77 @@
+import { useQuery, UseQueryResult, UseQueryOptions } from 'react-query';
+import wpcom from 'calypso/lib/wp'; // eslint-disable-line no-restricted-imports
+import { Design } from '../types';
+
+// Ideally this data should come from the themes API, maybe by a tag that's applied to
+// themes? e.g. `link-in-bio` or `no-fold`
+const STATIC_PREVIEWS = [
+	'bantry',
+	'sigler',
+	'miller',
+	'pollard',
+	'paxton',
+	'jones',
+	'baker',
+	'kingsley',
+];
+
+export interface UseThemeDesignsQueryOptions {
+	filter?: string;
+	tier?:
+		| '' // Returns all themes regardless of tier
+		| 'free'
+		| 'premium';
+}
+
+export function useThemeDesignsQuery(
+	{ filter = 'auto-loading-homepage', tier = '' }: UseThemeDesignsQueryOptions = {},
+	queryOptions: UseQueryOptions< unknown, unknown, Design[] > = {}
```

Comment:
IMO, I think `enabled` parameter is enough and we might not have no need to customize `queryOptions`.

Reply:
Accepting an entire `UseQueryOptions` object does make it pretty powerful. Perhaps it's exposing too much. But I took the cue from the query hooks the Calypso team have already added to the `/client/data` folder.
- `useWPCOMPlugins()`
- `useP2GuestsQuery()`
- `useSiteRolesQuery()`
- etc.

All accept a query options object. I think the idea is that as we use `react-query` more the API will become ubiquitous, and devs will expect to be able to use options like `enabled`, `onSuccess` or `refetchOnMount` anywhere.

So I think it comes down to `react-query` becoming a common way that we make network requests and so a pattern devs are expecting. Similar I guess to how React components often spread all of the props they receive down to their children, even though that makes the API super open and powerful. It's just a pattern that we are all accustomed to now.